testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_collections.py

```diff
@@ -0,0 +1,131 @@
+import uuid
+import pytest
+from CBLClient.Database import Database
+from CBLClient.Collection import Collection
+from CBLClient.Document import Document
+from CBLClient.Replication import Replication
+from libraries.testkit import cluster
+from keywords.ClusterKeywords import ClusterKeywords
+from libraries.testkit.admin import Admin
+from keywords import couchbaseserver
+from keywords.utils import log_info
+from keywords.MobileRestClient import MobileRestClient
+from keywords.constants import RBAC_FULL_ADMIN
+from keywords.SyncGateway import sync_gateway_config_path_for_mode
+from test_replication import verify_sgDocIds_cblDocIds
+
+bucket = "data-bucket"
+
+
+@pytest.fixture
+def teardown_doc_fixture():
+    def _delete_doc_if_exist(sg_client, url, db, doc_id, auth):
+        if sg_client.does_doc_exist(url, db, doc_id) is True:
+            sg_client.delete_doc(url, db, doc_id, auth=auth)
+    yield _delete_doc_if_exist
+
+
+@pytest.fixture(scope="function")
+def scope_collection_test_fixture(params_from_base_test_setup):
+    random_suffix = str(uuid.uuid4())[:8]
+    db_prefix = "db_"
+    scope_prefix = "scope_"
+    collection_prefix = "collection_"
+    cbl_db_name = db_prefix + random_suffix
+    scope = scope_prefix + random_suffix
+    collection = collection_prefix + random_suffix
+    need_sgw_admin_auth = params_from_base_test_setup["need_sgw_admin_auth"]
+    sg_username = "scope_collection_user" + random_suffix
+    log_info("****")
+    log_info(sg_username)
+    sg_password = "password"
+    data = {"bucket": bucket, "scopes": {scope: {"collections": {collection: {}}}}, "num_index_replicas": 0}
+    sg_db = "db"
+
+    sg_client = sg_url = sg_admin_url = None
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    mode = params_from_base_test_setup["mode"]
+    sg_config = sync_gateway_config_path_for_mode("sync_gateway_default_functional_tests", mode)
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_url = params_from_base_test_setup["sg_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
```

Comment:
This is makes the test run longer. Do we need to reset the cluster? Is it possible to avoid it? When I wrote the SGW scopes and collections tests I removed the reset and they seem to work.

Reply:
The conftest.py file sets the sync_gateway to a different mode than required. So this reset is necessary here.

Reply:
What mode is conftest.py is setting? Is it possible to change it there? Maybe we can move the test to to different location where conftest.py is in the right mode?

Reply:
The confest.py file sets it to travel-sample configuration, which won't allow us to create the database.
I will check once what else could be done instead of using reset.
Or we can just define a new contest for replication scopes and collections tests.